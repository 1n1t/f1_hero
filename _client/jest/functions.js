/* eslint-disable import/prefer-default-export */
export function sel(id) {
	return `[data-test="${id}"]`;
}
