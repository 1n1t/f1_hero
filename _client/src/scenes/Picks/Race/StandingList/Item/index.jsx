import React from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

const Item = ({ score, position, _driver }) => {
	const icon =
		score !== null ? (
			<div className="score">{score}pt</div>
		) : (
			<i className="fa fa-sort" aria-hidden="true" />
		);

	return (
		<div className="StandingItem">
			<div className="position">{position}.</div>
			<div className="driver-name">{_driver.name}</div>
			<div className="driver-code">{_driver.code}</div>
			{icon}
		</div>
	);
};

Item.defaultProps = {
	score: null
};

Item.propTypes = {
	score: PropTypes.number,
	position: PropTypes.number.isRequired,
	_driver: PropTypes.shape({
		name: PropTypes.string.isRequired,
		code: PropTypes.string.isRequired
	}).isRequired
};

export default Item;
