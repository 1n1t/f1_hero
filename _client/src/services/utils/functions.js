/* eslint-disable import/prefer-default-export */
export const isObjectEmpty = obj =>
	Object.keys(obj).length === 0 && obj.constructor === Object;

export const getNodePaddings = node => {
	const nodeStyle = window.getComputedStyle(node);
	return (
		parseInt(nodeStyle.getPropertyValue('padding-right'), 10) +
		parseInt(nodeStyle.getPropertyValue('padding-left'), 10)
	);
};
