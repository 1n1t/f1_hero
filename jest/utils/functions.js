const mongoose = require('mongoose');
const transformProps = require('transform-props');

function clearCollection(model) {
	return mongoose.model(model).remove();
}

function clearAllCollections() {
	Promise.all([
		mongoose.model('race').remove(),
		mongoose.model('user').remove(),
		mongoose.model('pick').remove(),
		mongoose.model('driver').remove()
	]).then(() => null);
}

function castToString(arg) {
	return String(arg);
}

function docToObj(docObj) {
	return transformProps(docObj, castToString, '_id');
}

module.exports = {
	clearCollection,
	clearAllCollections,
	docToObj
};
