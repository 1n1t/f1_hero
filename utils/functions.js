function getMaxFiveElements(arr) {
	return arr.sort((a, b) => a < b).slice(0, 5);
}

function capitalizeFirstLetter(string) {
	return string.charAt(0).toUpperCase() + string.slice(1);
}

module.exports = {
	getMaxFiveElements,
	capitalizeFirstLetter
};
