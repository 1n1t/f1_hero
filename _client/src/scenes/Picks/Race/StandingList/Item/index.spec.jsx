import r from 'Root/jest/utils/renderer';
import pickSample from 'JestClient/samples/pick';
import { sel } from 'JestClient/functions';

import Item from './index';

const defaultProps = { ...pickSample.forecast[0] };

r.init(Item, defaultProps);

it('renders correctly', () => {
	const wrapper = r.render(null, true);
	expect(wrapper).toMatchSnapshot();
});

describe('displays correct element', () => {
	it('when forecast was submitted', () => {
		const wrapper = r.render({ score: 5 }, true);
		const scoreEl = wrapper.find(sel('standing-list-item-score'));
		const sortIcon = wrapper.find(sel('standing-list-item-sort'));

		expect(scoreEl).toHaveLength(1);
		expect(sortIcon).toHaveLength(0);
	});

	it('when user is picking a new forecast', () => {
		const wrapper = r.render({ score: null }, true);
		const scoreEl = wrapper.find(sel('standing-list-item-score'));
		const sortIcon = wrapper.find(sel('standing-list-item-sort'));

		expect(scoreEl).toHaveLength(0);
		expect(sortIcon).toHaveLength(1);
	});
});
