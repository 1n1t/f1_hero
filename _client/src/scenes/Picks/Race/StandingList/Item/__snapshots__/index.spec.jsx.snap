// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="StandingItem"
>
  <div
    className="position"
  >
    1
    .
  </div>
  <div
    className="driver-name"
  >
    L. Hamilton
  </div>
  <div
    className="driver-code"
  >
    HAM
  </div>
  <div
    className="score"
    data-test="standing-list-item-score"
  >
    18
    pt
  </div>
</div>
`;
